src/coreComponents/physicsSolvers/multiphysics/MultiphasePoromechanicsKernel.hpp

```diff
@@ -174,61 +170,54 @@ class Multiphase :
   {
 public:
 
-    static constexpr int numDispDofPerElem =  Base::StackVariables::maxNumRows;
-
     /// Constructor.
     GEOSX_HOST_DEVICE
     StackVariables():
-      Base::StackVariables(),
-            xLocal(),
-            u_local(),
-            uhat_local(),
-            localResidualMomentum( Base::StackVariables::localResidual ),
-      dLocalResidualMomentum_dDisplacement( Base::StackVariables::localJacobian ),
-      dLocalResidualMomentum_dPressure{ {0.0} },
-      dLocalResidualMomentum_dComponents{ {0.0} },
-      localResidualMass{ 0.0 },
-      dLocalResidualMass_dDisplacement{ {0.0} },
-      dLocalResidualMass_dPressure{ {0.0} },
-      dLocalResidualMass_dComponents{ {0.0} },
-      localPoreVolumeConstraint{ 0.0 },
-      dLocalPoreVolumeConstraint_dPressure{ {0.0} },
-      dLocalPoreVolumeConstraint_dComponents{ {0.0} },
-      localPressureDofIndex{ 0 },
-      localComponentDofIndices{ 0 }
+      Base::StackVariables()
     {}
 
-#if !defined(CALC_FEM_SHAPE_IN_KERNEL)
-    /// Dummy
-    int xLocal;
-#else
-    /// C-array stack storage for element local the nodal positions.
-    real64 xLocal[numNodesPerElem][3];
-#endif
-
-    /// Stack storage for the element local nodal displacement
-    real64 u_local[numNodesPerElem][numDofPerTrialSupportPoint];
-
-    /// Stack storage for the element local nodal incremental displacement
-    real64 uhat_local[numNodesPerElem][numDofPerTrialSupportPoint];
-
-    real64 ( &localResidualMomentum )[numDispDofPerElem];
-    real64 ( &dLocalResidualMomentum_dDisplacement )[numDispDofPerElem][numDispDofPerElem];
-    real64 dLocalResidualMomentum_dPressure[numDispDofPerElem][1];
-    real64 dLocalResidualMomentum_dComponents[numDispDofPerElem][numMaxComponents];
-
-    real64 localResidualMass[numMaxComponents];
-    real64 dLocalResidualMass_dDisplacement[numMaxComponents][numDispDofPerElem];
-    real64 dLocalResidualMass_dPressure[numMaxComponents][1];
-    real64 dLocalResidualMass_dComponents[numMaxComponents][numMaxComponents];
-
-    real64 localPoreVolumeConstraint[1];
-    real64 dLocalPoreVolumeConstraint_dPressure[1][1];
-    real64 dLocalPoreVolumeConstraint_dComponents[1][numMaxComponents];
+    /// Mass accumulation
+    real64 compMassIncrement[maxNumComponents]{};
+    /// Derivative of mass accumulation wrt volumetric strain increment
+    real64 dCompMassIncrement_dVolStrainIncrement[maxNumComponents]{};
+    /// Derivative of mass accumulation wrt pressure
+    real64 dCompMassIncrement_dPressure[maxNumComponents]{};
+    /// Derivative of mass accumulation wrt comp density
+    real64 dCompMassIncrement_dComponents[maxNumComponents][maxNumComponents]{};
+
+    /// Pore volume constraint
+    real64 poreVolConstraint;
+    /// Derivative of pore volume constraint wrt pressure
+    real64 dPoreVolConstraint_dPressure;
+    /// Derivative of pore volume constraint wrt comp density
+    real64 dPoreVolConstraint_dComponents[1][maxNumComponents]{};
```

Comment:
Here, I left the 2-dimensional array, otherwise it becomes very difficult to use the generic functions in `BilinearFormUtilities`. Maybe this is something that can be improved later for cell-centered P0 quantities coming from the flow equations.